src/lib/reasoners/theory.ml

```diff
@@ -996,22 +888,35 @@ module Main_Default : S = struct
   let get_case_split_env t = t.gamma_finite
 
   let compute_concrete_model env =
-    let { gamma_finite; assumed_set; objectives; _ }, _ =
-      do_case_split_aux env ~for_model:true in
-    CC_X.extract_concrete_model
-      ~prop_model:assumed_set
-      ~optimized_splits:objectives
-      gamma_finite
+    if Options.get_interpretation () then
```

Comment:
I don't think this is the right place to have that test. Calling `compute_concrete_model` should always compute a model, but we shouldn't call `compute_concrete_model` if `Options.get_interpretation ()` is false (for similar reasons as #833 ).

Reply:
I bubble the test in `may_update_last_saved_model`. The current condition to trigger the computation of models is quite complicated. I'm not sure keeping the possibility of generate models at different stages of the SAT solver is very useful.